File: 1890-sum-of-beauty-of-all-substrings/sum-of-beauty-of-all-substrings.cpp
class Solution {
public:
    int beauty(vector<int> freq){
        int minf = INT_MAX;
        int maxf = 0;
        for(int i = 0 ; i < freq.size(); i++){
            if(freq[i] > 0){
                if(maxf < freq[i]){
                    maxf = freq[i];
                }
                if(minf > freq[i]){
                    minf = freq[i];
                }
            }
        }
        return maxf - minf;
    }
    int beautySum(string s) {
        int total = 0;
        for(int i = 0; i < s.size(); i++){
            vector<int> freq(26,0);
            for(int j = i; j < s.size(); j++){
                freq[s[j] - 'a']++;
                total += beauty(freq);
            }
            
        }
        return total;
    }
};

File: 1890-sum-of-beauty-of-all-substrings/sum-of-beauty-of-all-substrings.py
class Solution:
    def beauty(self, freq: List[int]) -> int:
        minFreq = float('inf')
        maxFreq = float('-inf')

        for i in range(len(freq)):
            if freq[i] > 0:
                if freq[i] < minFreq:
                    minFreq = freq[i]
                if freq[i] > maxFreq:
                    maxFreq = freq[i]
        
        return maxFreq - minFreq

    def beautySum(self, s: str) -> int:
        total = 0
        for i in range(len(s)):
            freq = [0]*26
            for j in range(i, len(s)):
                freq[ord(s[j]) - ord('a')] += 1
                total += self.beauty(freq)

        return total
